### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by AntonioHincapie**
- Set up associations between models.
- Write SQL queries with Active Record.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,17 @@
...
+  has_many :likes, foreign_key: 'posts_id'
+
+  def most_recent_comments
+    comments.limit(5).order('created_at desc')
+  end
+
+  after_save :update_counter
+
+  private
+
```
Comment (OLIPLICHE) [inline app/models/post.rb:13]: Nice Job for implementing the `after_save`  to call the `update_counter` method. 👍

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,2 +1,12 @@
...
+  belongs_to :user
+  belongs_to :post
+
+  after_save :update_counter
+
+  private
+
+  def update_counter
+    user.increment!(:PostCounter)
+  end
```
Comment (OLIPLICHE) [inline app/models/comment.rb]: - Please in reading the project requirements, it stated that "Comments has a method that updates the **comments counter** for a **post**".  Kindly check the code above and correct the mistake.

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @AntonioHincapie , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Git flow is correctly used (Pull Request is made from the feature branch into the dev branch) 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 2 above)
Comment (AntonioHincapie) [reply app/models/comment.rb]: Hi @OLIPLICHE,
Thanks for your feedback.
The change that you required it's already solved.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: 'users_id'
+  has_many :likes, foreign_key: 'users_id'
+  has_many :comments, foreign_key: 'users_id'
```
Comment (medaminedev66) [inline app/models/user.rb]: [OPTIONAL] In your DB schema, there is no users_id column in the posts, likes, and comments tables. I recommend fixing the foreign_keys at the line (2,3 and 4) so the associations between models is correct.

File: app/models/post.rb [hunk 4]
```diff
@@ -1,2 +1,17 @@
 class Post < ApplicationRecord
+  belongs_to :user
```
Comment (medaminedev66) [inline app/models/post.rb]: [OPTIONAL] According to your DB schema, this line should be:
```rb
belongs_to :AuthorId, class_name: 'User'
```
Check out this article to learn more about that: [Use a custom column name with a belongs_to association in rails](https://stevepolito.design/blog/use-a-custom-column-name-with-a-belongs-to-association-in-rails/)

**Review by medaminedev66 — APPROVED** [review]
Hi @AntonioHincapie 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- There are no linters errors.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**